Gems/Atom/RHI/Code/Include/Atom/RHI/CpuProfilerImpl.h

```diff
@@ -129,6 +131,15 @@ namespace AZ
             AZStd::vector<RHI::Ptr<CpuTimingLocalStorage>, AZ::OSStdAllocator> m_registeredThreads;
             AZStd::mutex m_threadRegisterMutex;
 
+            // Storage for runtime GroupRegionNames
+            using DynamicRegionNameMap = AZStd::unordered_map<AZStd::string, TimeRegion::GroupRegionName>;
+             
+            // Use string_view over the const char* as a key because equivalent string literals throughout the source files might be duplicated
+            // (and therefore have different addresses) if string pooling is disabled.
+            AZStd::unordered_map<AZStd::string_view, DynamicRegionNameMap> m_dynamicGroupRegionNameMap;
```

Comment:
Sure - the reason why I had originally implemented it this way is because this might lead to duplicate allocations for regions with different source locations but the same group name and region name. Granted, the cost of having these nested `unordered_maps` is probably much higher than the cost of allocating two pointers here so I fixed it. Had to override `operator==` and implement a hash to do it, though.